crates/matrix-sdk/tests/integration/matrix_auth.rs

```diff
@@ -313,3 +323,292 @@ fn test_serialize_session() {
         })
     );
 }
+
+#[cfg(feature = "e2e-encryption")]
+#[async_test]
+async fn test_login_with_cross_signing_bootstrapping() {
```

Comment:
Can we also test the case where the bootstrap fails, so we know that a bootstrap failure won't tun into a login failure.

Reply:
So I removed the login failure if bootstrapping failed; although I am still of the opinion that if one requests automatic cross-signing bootstrapping upon login, then they should be made aware if that failed; a warning is just so easy to miss and ignore).

That being said: I've added a test for that, and we should discuss it in private.